### ISSUE TITLE
Flatpak 1.11.1 breaks passing files through portal in certain apps

### ISSUE THREAD

**Issue by Erick555** [flatpak/flatpak#4308]
## Linux distribution and version
ArchLinux
## Flatpak version
1.11.1
## Description of the problem
Since flatpak 1.11.1 it's impossible to pass files through portal to 3rd party when flatpak apps sets `TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"` (note that official docs [recommend](https://docs.flatpak.org/en/latest/sandbox-permissions.html#filesystem-access) that if needed). This may be caused by [symlinking](https://github.com/flatpak/flatpak/commit/40510e8ae8fd433df04959938b8e81e582849481) `$XDG_RUNTIME_DIR/app` to `../../flatpak/app`.
## Steps to reproduce

1. Install flatpak 1.11.1
2. Install  [Thunderbird flatpak](https://flathub.org/apps/details/org.mozilla.Thunderbird)
3. Try to open email attachment in external app

**Comment by smcv** [comment]
Is `TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"` being set by the Thunderbird Flatpak app?

Is your external app running on the host OS?

**Comment by Erick555** [comment]
@smcv yes [thunderbird sets it](https://github.com/flathub/org.mozilla.Thunderbird/blob/master/org.mozilla.Thunderbird.json#L217). The external app is supposed to be launched through portal on host os but the portal dialog doesn't even show up.

**Comment by Erick555** [comment]
If I change it to `export TMPDIR=$XDG_CACHE_HOME/tmp` like [ff does](https://hg.mozilla.org/integration/autoland/file/tip/taskcluster/docker/firefox-flatpak/launch-script.sh#l2). It works. You may try to switch ff script to use the `TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"` line and reproduce it in ff.

**Comment by smcv** [comment]
xdg-desktop-portal needs updating to catch up with Flatpak.
